src/algebra/ring/basic.lean

```diff
@@ -257,6 +258,29 @@ theorem coe_monoid_hom_injective : function.injective (coe : (α →+* β) → (
 /-- Ring homomorphisms preserve multiplication. -/
 @[simp] lemma map_mul (f : α →+* β) (a b : α) : f (a * b) = f a * f b := f.map_mul' a b
 
+/-- `f : R →+* S` has a trivial codomain iff `f 1 = 0`. -/
+lemma codomain_trivial_iff_map_one_eq_zero : (0 : β) = 1 ↔ f 1 = 0 :=
+by rw [map_one, eq_comm]
+
+/-- `f : R →+* S` has a trivial codomain iff it has a trivial range. -/
+lemma codomain_trivial_iff_range_trivial : (0 : β) = 1 ↔ (∀ x, f x = 0) :=
+f.codomain_trivial_iff_map_one_eq_zero.trans
+  ⟨λ h x, by rw [←mul_one x, map_mul, h, mul_zero], λ h, h 1⟩
+
+/-- `f : R →+* S` has a trivial codomain iff its range is `{0}`. -/
+lemma codomain_trivial_iff_range_eq_singleton_zero : (0 : β) = 1 ↔ set.range f = {0} :=
+f.codomain_trivial_iff_range_trivial.trans
+  ⟨ λ h, set.ext (λ y, ⟨λ ⟨x, hx⟩, by simp [←hx, h x], λ hy, ⟨0, by simpa using hy.symm⟩⟩),
+    λ h x, set.mem_singleton_iff.mp (h ▸ set.mem_range_self x)⟩
+
+/-- `f : R →+* S` doesn't map `1` to `0` if `S` is nontrivial -/
+lemma map_one_ne_zero_of_nontrivial [nontrivial β] : f 1 ≠ 0 :=
```

Comment:
I don't think that we should mention `of_nontrivial` in lemma names. E.g., we avoid it in `zero_ne_one`. Why should we mention it in `ring_hom.map_one_ne_zero`? Same for `map_monic_ne_zero`.

Otherwise LGTM.